### PR TITLE
V2/fetch ens

### PR DESCRIPTION
**Pull request by dospore**
Fetch and set ENS using an ENS store with a default L1 provider. Update README to include this env var.

Would like to get thoughts on TTL and cache vs localStorage

### CODE REVIEW

File: store/ENSSlice/index.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const getCachedENS = (account: string): string | null => {
+    const nameStr = localStorage.getItem(ethers.utils.getAddress(account));
+    if (!nameStr) {
+        return null;
+    }
+    const ensName = JSON.parse(nameStr);
+    const now = Date.now();
+    if (now > ensName.expiry) {
+        localStorage.removeItem(ethers.utils.getAddress(account));
+        return null;
```
Comment (ScaredIbis) [inline store/ENSSlice/index.ts:25]: could potentially simplify here by making this function set the ens name in the cache if its expired or does not exist and not needing `storeENSName`

Comment (dospore) [reply store/ENSSlice/index.ts:25]: not entirely sure what you mean by this? restoring if its expired?

Comment (ScaredIbis) [reply store/ENSSlice/index.ts:25]: yeah pretty much just putting the logic of `storeENSName` here and returning the result. 

so the behaviour of `getCachedENS` is actually more like `get the cached one or populate the cache if required and return the result`

Comment (dospore) [reply store/ENSSlice/index.ts:25]: but then say no cached items exist id have to
getCached -> provider.lookupAddress -> getCached or always provider.lookupAddress -> getCached

**Review by ScaredIbis — APPROVED** [review]
lgtm, works as expected

**Pull request merged**